end-to-end-tests/echo_test.go

```diff
@@ -0,0 +1,205 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+//go:build endtoendtests
+// +build endtoendtests
+
+package endtoendtests
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/cartesi/rollups-node/internal/deps"
+	"github.com/cartesi/rollups-node/internal/machine"
+	"github.com/cartesi/rollups-node/internal/node"
+	"github.com/cartesi/rollups-node/pkg/readerclient"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	payload              = "0xdeadbeef"
+	maxReadInputAttempts = 10
+	defaulTxDatabaseFile = "default_tx_database"
+)
+
+type EchoInputTestSuite struct {
+	suite.Suite
+	containers    *deps.DepsContainers
+	cancel        context.CancelFunc
+	tempDir       string
+	supervisorErr chan error
+}
+
+func (s *EchoInputTestSuite) SetupSuite() {
+
+	// Clear default_tx_database
+	cwd, err := os.Getwd()
+	require.NoError(s.T(), err)
+
+	_ = os.Remove(filepath.Join(cwd, defaulTxDatabaseFile))
+
+	// Create machine snapshot
+	tempDir, err := ioutil.TempDir("", "machine-snapshot")
+	assert.NoError(s.T(), err)
+	machine.Save("cartesi/rollups-node-snapshot:devel", tempDir, "test-echo-app")
+
+	// Run deps
+	ctx := context.Background()
+	var depsConfig = deps.NewDefaultDepsConfig()
+	depsContainers, err := deps.Run(ctx, *depsConfig)
+	require.NoError(s.T(), err)
+
+	// Wait a little for Dependencies to be actually ready
+	time.Sleep(2 * time.Second)
+
+	// Fix the Blochain timestamp. Must be "in the future"
+	err = SetNextBlockTimestamp(ctx, LocalBlockchainHttpEndpoint, 7000000000)
+	require.NoError(s.T(), err)
+
+	// Run Node Service
+	nodeConfig := NewLocalNodeConfig()
+
+	nodeConfig.SnapshotDir = tempDir
+
+	ctx, cancel := context.WithCancel(ctx)
+	supervisor, err := node.Setup(ctx, nodeConfig)
+	require.NoError(s.T(), err)
+
+	ready := make(chan struct{}, 1)
+	supervisorErr := make(chan error, 1)
+	go func() {
+		err := supervisor.Start(ctx, ready)
+		if err != nil {
+			supervisorErr <- err
+		}
+	}()
+
+	select {
+	case err := <-supervisorErr:
+		require.NoError(s.T(), err)
+	case <-ready:
+		break
+	}
+
+	// Configure Suite for tear down
+	s.containers = depsContainers
+	s.tempDir = tempDir
+	s.cancel = cancel
+	s.supervisorErr = supervisorErr
+
+}
+
+func (s *EchoInputTestSuite) TearDownSuite() {
+
+	// Stop Node services
+	s.cancel()
+
+	// Remove machine snpshot
+	os.RemoveAll(s.tempDir)
+
+	// Terminate deps
+	ctx := context.Background()
+	err := deps.Terminate(ctx, s.containers)
+	require.NoError(s.T(), err)
+
+	// Clear default_tx_database
+	cwd, err := os.Getwd()
+	require.NoError(s.T(), err)
+
+	_ = os.Remove(filepath.Join(cwd, defaulTxDatabaseFile))
+
+}
+
+func (s *EchoInputTestSuite) SetupTest() {
+
+}
+
+func (s *EchoInputTestSuite) TestSendInput() {
+
+	ctx := context.Background()
```

Comment:
We should create a WithTimeout context to avoid the chance of having the test hanging.

Reply:
The Go Test itself has a timeout feature. The default is 10 minutes.

Reply:
Made the context to have timeout also, despite the tests already having a timeout feature